libs/sdk-model/src/execution/filter/filterMerge.ts

```diff
@@ -62,17 +45,19 @@ function separateFiltersByType(filters: IFilter[]): FilterByType {
 /**
  * Merges two sets of filters.
  *
- * - Attribute from both sets are simply concatenated => bunch of ANDed filters
+ * - Attribute filters and measure value filters from both sets are simply concatenated resulting
+ *   in the filters being ANDed together.
  * - Date filters are merged based on date data set they filter on
- * - For Date filters for the same date data set:
- *   -  filters from the addedFilters list override filters in the original list
+ *   - For Date filters for the same date data set:
+ *     - the filters are ordered putting original filters first
+ *     - the last filter in this ordering is taken
+ *        - if it is All time, all filters for the dimension are cleared
+ *        - else the last filter is used
  *
- * TODO: we seem to be missing the original logic that was cleaning up filters if the new filter was ALL_TIME?
- *
- * TODO: this logic is not in the right place; filter merging needs to be done by backend implementation
- *  to be able to correctly identify objects being filtered on - regardless of how they are referenced. this
- *  function will return bogus if one filter references say display form by URI and the other references
- *  the same display form by identifier.
+ * @remarks
+ * It is the responsibility of the caller to make sure all the filters use the same ObjRef type so that
+ * they can be compared without involving the backend. Otherwise, the results might be unexpected
+ * (especially for date filters).
```

Comment:
please enhance this with a note that the backend's insight service has a method to help in particular use-cases

Reply:
Added, good point :)